apps/designer/app/designer/designer.tsx

```diff
@@ -268,6 +300,14 @@ export const Designer = ({ config, project, page }: DesignerProps) => {
     [publishRef, onRefReadCanvasWidth, onRefReadCanvas]
   );
 
+  const page = useMemo(() => {
```

Comment:
I feel like you will need this in many places, maybe instead of placing pageId into nano state it should be the page

Reply:
Yeah, not sure. This is an intermediary PR so a lot of small things like this are still in flux, will be easier to tell which way is better later.